contracts/weapons.sol

```diff
@@ -379,52 +382,92 @@ contract Weapons is Initializable, ERC721Upgradeable, AccessControlUpgradeable,
         return result;
     }
 
+    function getDustSupply(uint8 stars) public view returns (uint16) {
+        return uint16(burnDust[msg.sender] >> (stars * 16));
+    }
+
+    function burn(uint256 burnID) public restricted {
+        Weapon storage burning = tokens[burnID];
+        uint8 stars = getStarsFromProperties(burning.properties);
+        require(getDustSupply(stars) < 65535, "Dust supply capped");
+        _burn(burnID);
+        burnDust[msg.sender] += (uint256(1) << (stars * 16));
```

Comment:
Please move this and the `require()` statement just above that checks dust supply, to a dedicated `internal` function called something like `_incrementDustSupply(address playerAddress, uint8 stars)`. We want to make sure to encapsulate working with burn dust, especially when bit operations like this are involved (yes, even if the code seems simple).